### PR TITLE
difftest: writeback support for dual core 

### PR DESCRIPTION
**Pull request by xiaokamikami**
Multicore support for hasdut added on Paladin or VCS

### CODE REVIEW

File: src/main/scala/Gateway.scala [hunk 1]
```diff
@@ -249,14 +250,14 @@ class GatewaySinkControl(config: GatewayConfig) extends Bundle {
...
-    val module = Module(new Preprocess(chiselTypeOf(bundles).toSeq, config))
+  def apply(bundles: MixedVec[DifftestBundle], config: GatewayConfig, numcores: Int): MixedVec[DifftestBundle] = {
+    val module = Module(new Preprocess(chiselTypeOf(bundles).toSeq, config, numcores))
     module.in := bundles
     module.out
   }
 }
 
-class Preprocess(bundles: Seq[DifftestBundle], config: GatewayConfig) extends Module {
+class Preprocess(bundles: Seq[DifftestBundle], config: GatewayConfig, numcores: Int) extends Module {
```
Comment (poemonsense) [inline src/main/scala/Gateway.scala:260]: No need to pass `numcore` to this class.

Just add `val numCores = bundles.count(_.isUniqueIdentifier)` in this class, and reuse this `numCores` in this class.

File: src/main/scala/Gateway.scala [hunk 2]
```diff
@@ -265,13 +266,15 @@ class Preprocess(bundles: Seq[DifftestBundle], config: GatewayConfig) extends Mo
   if (config.hasDutZone || config.isSquash || config.isBatch) {
     // Special fix for int writeback. Work for single-core only
     if (in.exists(_.desiredCppName == "wb_int")) {
-      require(in.count(_.isUniqueIdentifier) == 1, "only single-core is supported yet")
+      if (config.isSquash || config.isBatch) {
+        require(in.count(_.isUniqueIdentifier) == 1, "only single-core is supported yet")
```
Comment (poemonsense) [inline src/main/scala/Gateway.scala:270]: This line is the same as `require(numCores == 1, ......)`

**Review by klin02 — APPROVED** [review]

**Review by linkedlist771 — COMMENTED** [review]
I noticed a mix of camelCase (e.g., `numPhyRegs`, `wb_int`) and snake_case (e.g., `wb_for_skip`) naming conventions. To improve consistency and readability, I recommend unifying the naming style throughout the code. In Scala, the generally preferred style for variables and methods is camelCase.

Suggestions:

- Rename `wb_for_skip` to `wbForSkip`.
-  ....

This minor change will help maintain a cohesive code style and make the code easier to understand for future contributors.

**Comment by linkedlist771** [comment]
I have created a [PR](https://github.com/OpenXiangShan/difftest/pull/384)  about this issue.😊

**Pull request merged**